sentry-opentelemetry/spec/sentry/opentelemetry/span_processor_spec.rb

```diff
@@ -154,6 +154,20 @@
 
     it 'starts a sentry transaction on otel root span' do
       expect(Sentry).to receive(:start_transaction).and_call_original
+      expect_any_instance_of(Sentry::Transaction).to receive(:set_initial_sample_decision).with(
+        sampling_context: a_hash_including(
+          otel: {
+            attributes: root_span.attributes,
+            resource: root_span.resource.attribute_enumerator.to_h,
+            kind: root_span.kind,
+            instrumentation_scope: {
+              name: root_span.instrumentation_scope.name,
+              version: root_span.instrumentation_scope.version
+            }
+          }
+        )
+      )
```

Comment:
It would be good to not rely on mocking here, especially that it touches internals. Would you be able to improve this spec, or maybe cover this change in another spec where mocks are not needed?

Reply:
Addressed [here](https://github.com/getsentry/sentry-ruby/pull/2683/commits/7c8ab68175b91edaab8db9ddeab5293445f6c367)